docs/content/languages/nodejs.md

```diff
@@ -0,0 +1,420 @@
+*   Trying 0.0.0.0
+.
+.
+.
+< HTTP/1.1 200 OK
+< Content-Length: 12
+< Content-Type: text/html; charset=utf-8
+< Date: Tue, 23 Oct 2018 19:01:55 GMT
+< Etag: W/"c-QwzjTQIHJO11oZbfwq1nx3dy0Wk"
+< X-Fission-Function-Name: header-example
+< X-Fission-Function-Namespace: default
+< X-Fission-Function-Resourceversion: 19413500
+< X-Fission-Function-Uid: 0014884b-d6e7-11e8-afb7-42010a800194
+< X-Powered-By: Express
+< 
+* Connection #0 to host 0.0.0.0 left intact
+hello world!
+```
+
+#### Setting Status Codes 
+
+Create a file error-handling.js with the following content. Here the function tries to validate an input parameter "job_id" and sends a HTTP response code 400 when validation fails.   
+
+```
+module.exports = async function(context) {
+    const stringBody = JSON.stringify(context.request.body);
+    const body = JSON.parse(stringBody);
+    console.log(body);
+
+    const job = body.job_id;
+    const jobStatus = body.job_status;
+
+    console.log("Received CI job id: " + job + " job status: " + jobStatus );
+
+    if (!job) {
+        return {
+            status: 400,
+            body: "job_id cannot be empty"
+        };
+    }
+
+    return {
+        status: 200,
+        body: "Successfully saved CI job status for job ID: " + job
+    };
+}
+```
+
+Create a function with the following command. 
+
+```
+fission function create --name error-handling --code error-handling.js --env nodejs
+```
+
+Create an http trigger to invoke the function
+
+```
+fission httptrigger create --url /error-handling --function error-handling --method POST 
+```
+
+Invoke the function with this curl command where job_id is empty and you should see "job_id cannot be empty"
+
+```
+curl -XPOST http://$FISSION_ROUTER/error-handling -d '{"job_status": "Passed"}'
+```
+
+## Working with dependencies
+
+There may be instances where functions need to require node modules that are not packed into the nodejs runtime environment. In such instances, nodejs builder image could be used to `npm install` those modules.
+This section describes ways in which this can be achieved.
+
+### Using fission nodejs builder image
+
+#### Example of using the nodejs builder image
+
+fission docker hub has a nodejs builder image `fission/node-builder`. Here's an example of using this image.
+
+First, create an environment with runtime image and builder image as follows 
+ 
+```
+fission environment create --name nodejs --image fission/node-env --builder fission/node-builder
+```
+
+Next, create a file moment-example.js with the following content. This file requires 'moment' node_module that is not packed into the fission runtime image. Also create a package.json with 'moment' listed in dependencies section.
+
+```
+const momentpackage = require('moment')
+
+module.exports = async function(context) {
+
+    return {
+        status: 200,
+        body: momentpackage().format()
+    };
+}  
+```
+
+Next, create a zip archive of these 2 files, let's call it node-source-example.zip
```

Comment:
Better to show directory structure and command for arching. In case that users archive directory with the wrong path or structure.